### PR TITLE
Add custom pod fake logs feature

### LINKED ISSUE #96957: Custom logging message for fake pod GetLogs in go-client
**What would you like to be added**:

I would like a way to have a custom string be retrieved from the fake pod GetLogs method.

**Why is this needed**:

This is needed because currently the fake GetLogs method always results in a log message "fake logs", and it would be better for testing if a log message set by the user could be retrieved instead.

(This would probably involve adding a portion to the fake container where the custom logs can be placed)

### PR DESCRIPTION
**Pull request by yuzhiquan**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Add custom pod fake logs feature
**Which issue(s) this PR fixes**:

Fixes #96957

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by yuzhiquan** [comment]
It's likely should not edit fak_pod diractly, so how to add new implement in fakepod struct?

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_pod_expansion.go [hunk 1]
```diff
@@ -93,3 +101,16 @@ func (c *FakePods) Evict(ctx context.Context, eviction *policy.Eviction) error {
 func (c *FakePods) ProxyGet(scheme, name, port, path string, params map[string]string) restclient.ResponseWrapper {
 	return c.Fake.InvokesProxy(core.NewProxyGetAction(podsResource, c.ns, scheme, name, port, path, params))
 }
+
+// SetContainerFakeLogs set fake logs for specified container and log content
+func (c *FakePods) SetContainerFakeLogs(container, fakeLogs string) error {
+	if container == "" {
```
Comment (lingsamuel) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_pod_expansion.go:107]: Do we only call this function in a trusted codepath? If so, I don't think we need a empty check.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_pod_expansion_test.go [hunk 2]
```diff
@@ -31,21 +31,48 @@ func TestFakePodsGetLogs(t *testing.T) {
...
-		t.Fatal("Empty log")
+
+	testData := []struct {
+		pod       string
+		container string
+		logs      string
+	}{
+		{
+			pod:       "foo",
+			logs:      defaultFakeLogs,
```
Comment (lingsamuel) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_pod_expansion_test.go:42]: fmt this

**Pull request closed without merging**